src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -231,6 +231,18 @@
   (env-config-for-db-ulong "PDB_COMMAND_SQL_STATEMENT_TIMEOUT_MS"
                            (* 10 60 1000)))
 
+(def table-lock-order
+  "This provides a partial ordering of tables, to ensure that we always lock
+  them in a consistent order."
+  ["certnames" "reports" "resource_events"])
+
+(defn acquire-locks!
+  "Acquires a lock on the specified table."
+  [table-modes]
+  {:pre [(every? string? (keys table-modes))
+         (every? string? (vals table-modes))]}
+  (let [ordered-tables (sort-by #(.indexOf table-lock-order %) (keys table-modes))]
```

Comment:
This will put tables that aren't covered by the ordering at the start (`.indexOf == -1`) which seems vacuously correct.